blobs.go

```diff
@@ -92,6 +92,10 @@ type BlobStatter interface {
 	Stat(ctx context.Context, dgst digest.Digest) (Descriptor, error)
 }
 
+type BlobAccessFactory interface {
+	BlobAccessController(svc BlobDescriptorService) BlobDescriptorService
```

Comment:
`Clear` removes access to a blob from a repository.  Calling this should make `Stat` return false, but it doesn't remove any blobs from the blob store.  Does your custom `BlobDescriptorService` implement `Stat` based on it's metadata implementation?